packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -165,10 +165,12 @@ class FlutterManifest {
   ///   licenses:
   ///     - assets/foo_license.txt
   /// ```
-  List<String> get additionalLicenses => <String>[
-    if (_flutterDescriptor case {'licenses': final YamlList list})
-      for (final dynamic item in list) '$item',
-  ];
+  List<String> get additionalLicenses {
+    return <String>[
+      if (_flutterDescriptor case {'licenses': final YamlList list})
+        for (final Object? item in list) '$item',
+    ];
+  }
```

Comment:
Personally, I don't feel that the block body makes it easier to read, but I'm really glad that we caught the `dynamic` 👍

Reply:
(just gonna resolve this one since I'm fine with how it is)